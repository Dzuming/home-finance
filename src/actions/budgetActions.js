import * as types from './actionTypes';
import { createAction } from 'redux-actions';
import env from '../../environments/config';

export const fetchBudget = () => (dispatch, getState) => {
  const { id } = getState().user;
  return dispatch({
    type: types.API_REQUEST_GET,
    payload: {
      url: `${env.api_url}/api/budget/${id}`,
      success: setBudget,
    },
  });
};

export const setBudget = createAction(types.BUDGET_CRUD.READ);
