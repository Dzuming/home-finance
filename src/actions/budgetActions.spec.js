import * as actions from './budgetActions';
import expect from 'expect';

const budget = {
  budget: 50000,
};

describe('budget actions', () => {
  it('should create API_REQUEST_PENDING action with snapshot', () => {
    expect(actions.setBudget(budget)).toMatchSnapshot();
  });
});
