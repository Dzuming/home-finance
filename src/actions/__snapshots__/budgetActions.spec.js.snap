// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`budget actions should create API_REQUEST_PENDING action with snapshot 1`] = `
Object {
  "payload": Object {
    "budget": 50000,
  },
  "type": "READ_BUDGET",
}
`;
